### PR TITLE
[Modules] PSRule - Fix Rule Azure.Resource.UseTags

### LINKED ISSUE #2666: [PSRule] Fix Rule Azure.Resource.UseTags
### Rule Azure.Resource.UseTags
Error example: Error: AZR-000166: ***splhcom001 failed Azure.Resource.UseTags. Azure resources should be tagged using a standard convention.
There are no modules where we need to add tags support. So the 2 scenarios are:

- resource supporting tags but tags not used in test files
-  Azure resource not supporting tags

- [x] Add tags with values in test files for the following modules:
Microsoft.Compute/availabilitySets
Microsoft.Compute/diskEncryptionSets
Microsoft.Compute/disks
Microsoft.Compute/galleries
Microsoft.Compute/virtualMachines
Microsoft.Compute/virtualMachineScaleSets
Microsoft.ContainerInstance/containerGroups
Microsoft.DataProtection/backupVaults
Microsoft.HealthBot/healthBots
microsoft.insights/actionGroups
Microsoft.Insights/activityLogAlerts
Microsoft.Insights/components
Microsoft.Insights/metricAlerts
microsoft.insights/privateLinkScopes
Microsoft.Insights/scheduledQueryRules
Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies
Microsoft.Network/applicationSecurityGroups
Microsoft.Network/connections
Microsoft.Network/ddosProtectionPlans
Microsoft.Network/dnsResolvers
Microsoft.Network/firewallPolicies
Microsoft.Network/ipGroups
Microsoft.Network/loadBalancers
Microsoft.Network/localNetworkGateways
Microsoft.Network/networkInterfaces
Microsoft.Network/networkManagers
Microsoft.Network/networkWatchers
Microsoft.Network/privateDnsZones
Microsoft.Network/privateEndpoints
Microsoft.Network/privateLinkServices
Microsoft.Network/publicIPPrefixes
Microsoft.Network/routeTables
Microsoft.Network/virtualHubs
Microsoft.Network/virtualNetworks
Microsoft.Network/virtualWans
Microsoft.Network/vpnGateways
Microsoft.Resources/deploymentScripts
Microsoft.Resources/resourceGroups
Microsoft.ServiceFabric/clusters
Microsoft.SignalRService/webPubSub
Microsoft.Sql/servers
Microsoft.Storage/storageAccounts
Microsoft.Synapse/privateLinkHubs
Microsoft.VirtualMachineImages/imageTemplates
Microsoft.Web/connections
Microsoft.Web/staticSites
https://github.com/Azure/ResourceModules/pull/2241/files can be used as a reference, the following code needs to be added:
```
  tags: {
      Environment: 'Non-Prod'
      Role: 'DeploymentValidation'    ----> there is no standard with TAGS, should we standardize?
    }
```
Examples of other found tags:
tags: {      purpose: 'test'    }
- [x] Add resource or file path to exclusion because tags are not supported by the resource:
Microsoft.KubernetesConfiguration/extensions
Microsoft.KubernetesConfiguration/fluxConfigurations
Microsoft.ManagedServices/registrationAssignments
Microsoft.ManagedServices/registrationDefinitions
Microsoft.Management/managementGroups
Microsoft.OperationsManagement/solutions
Microsoft.PolicyInsights/remediations

### PR DESCRIPTION
**Pull request by fabmas**
# Description
[PSRule] Fix Rule Azure.Resource.UseTags 
close #2666 

| Pipeline |
| - |
| [.Platform: Library PSRule pre-flight validation](https://github.com/Azure/ResourceModules/actions/workflows/platform.librarycheck.psrule.yml)|

# Type of Change
- [x] New feature (non-breaking change which adds functionality)

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep [hunk 1]
```diff
@@ -158,6 +158,11 @@ module testDeployment '../../deploy.bicep' = {
       }
     ]
     systemAssignedIdentity: true
+<<<<<<< HEAD
```
Comment (eriqua) [inline modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep:161]: Please remove line
```suggestion
```

File: modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep [hunk 2]
```diff
@@ -158,6 +158,11 @@ module testDeployment '../../deploy.bicep' = {
...
+    tags: {
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+=======
```
Comment (eriqua) [inline modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep:165]: Please remove line
```suggestion
```

File: modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep [hunk 3]
```diff
@@ -215,6 +220,7 @@ module testDeployment '../../deploy.bicep' = {
           }
         }
       ]
+>>>>>>> 256b4b10ef1f210ecc80876c2d90dcdc5125d3bf
```
Comment (eriqua) [inline modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep:223]: Please remove line
```suggestion
```

File: modules/Microsoft.DBforPostgreSQL/flexibleServers/.test/private/dependencies.bicep [hunk 4]
```diff
@@ -49,6 +49,7 @@ resource privateDNSZone 'Microsoft.Network/privateDnsZones@2020-06-01' = {
                 id: virtualNetwork.id
             }
             registrationEnabled: false
+
```
Comment (eriqua) [inline modules/Microsoft.DBforPostgreSQL/flexibleServers/.test/private/dependencies.bicep:52]: Please remove line
```suggestion
```

File: modules/Microsoft.DesktopVirtualization/hostpools/.test/common/deploy.test.bicep [hunk 5]
```diff
@@ -100,6 +100,11 @@ module testDeployment '../../deploy.bicep' = {
         ram: 8
       }
     }
+<<<<<<< HEAD
```
Comment (eriqua) [inline modules/Microsoft.DesktopVirtualization/hostpools/.test/common/deploy.test.bicep:103]: Please remove line
```suggestion
```

File: modules/Microsoft.DesktopVirtualization/hostpools/.test/common/deploy.test.bicep [hunk 6]
```diff
@@ -100,6 +100,11 @@ module testDeployment '../../deploy.bicep' = {
...
+    tags: {
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+=======
```
Comment (eriqua) [inline modules/Microsoft.DesktopVirtualization/hostpools/.test/common/deploy.test.bicep:107]: Please remove line
```suggestion
```

File: modules/Microsoft.DesktopVirtualization/hostpools/.test/common/deploy.test.bicep [hunk 7]
```diff
@@ -114,6 +119,7 @@ module testDeployment '../../deploy.bicep' = {
           dayOfWeek: 'Saturday'
         }
       ]
+>>>>>>> 256b4b10ef1f210ecc80876c2d90dcdc5125d3bf
```
Comment (eriqua) [inline modules/Microsoft.DesktopVirtualization/hostpools/.test/common/deploy.test.bicep:122]: Please remove line
```suggestion
```

File: modules/Microsoft.DesktopVirtualization/workspaces/.test/common/deploy.test.bicep [hunk 8]
```diff
@@ -81,7 +81,16 @@ module testDeployment '../../deploy.bicep' = {
         principalType: 'ServicePrincipal'
       }
     ]
+<<<<<<< HEAD
```
Comment (eriqua) [inline modules/Microsoft.DesktopVirtualization/workspaces/.test/common/deploy.test.bicep:84]: Please remove line
```suggestion
```

File: modules/Microsoft.DesktopVirtualization/workspaces/.test/common/deploy.test.bicep [hunk 9]
```diff
@@ -81,7 +81,16 @@ module testDeployment '../../deploy.bicep' = {
...
+    workspaceDescription: 'This is my first AVD Workspace'
+    workspaceFriendlyName: 'My first AVD Workspace'
+    tags: {
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
+=======
```
Comment (eriqua) [inline modules/Microsoft.DesktopVirtualization/workspaces/.test/common/deploy.test.bicep:91]: Please remove line
```suggestion
```

File: modules/Microsoft.DesktopVirtualization/workspaces/.test/common/deploy.test.bicep [hunk 10]
```diff
@@ -81,7 +81,16 @@ module testDeployment '../../deploy.bicep' = {
...
     description: 'This is my first AVD Workspace'
     friendlyName: 'My first AVD Workspace'
+>>>>>>> 256b4b10ef1f210ecc80876c2d90dcdc5125d3bf
```
Comment (eriqua) [inline modules/Microsoft.DesktopVirtualization/workspaces/.test/common/deploy.test.bicep:94]: Please remove line
```suggestion
```

File: modules/Microsoft.DesktopVirtualization/workspaces/.test/common/deploy.test.bicep [hunk 11]
```diff
@@ -81,7 +81,16 @@ module testDeployment '../../deploy.bicep' = {
         principalType: 'ServicePrincipal'
       }
     ]
+<<<<<<< HEAD
+    workspaceDescription: 'This is my first AVD Workspace'
```
Comment (eriqua) [inline modules/Microsoft.DesktopVirtualization/workspaces/.test/common/deploy.test.bicep:85]: We have twice description and friendly name, the outdated ones should be removed

File: modules/Microsoft.KubernetesConfiguration/extensions/.test/common/deploy.test.bicep [hunk 12]
```diff
@@ -59,6 +59,7 @@ module testDeployment '../../deploy.bicep' = {
     releaseNamespace: 'flux-system'
     releaseTrain: 'Stable'
     version: '0.5.2'
+
```
Comment (eriqua) [inline modules/Microsoft.KubernetesConfiguration/extensions/.test/common/deploy.test.bicep:62]: Please remove added line
```suggestion
```

File: modules/Microsoft.Network/expressRouteCircuits/.test/common/deploy.test.bicep [hunk 13]
```diff
@@ -80,6 +80,9 @@ module testDeployment '../../deploy.bicep' = {
     ]
     skuFamily: 'MeteredData'
     skuTier: 'Standard'
-    allowClassicOperations: true
```
Comment (eriqua) [inline modules/Microsoft.Network/expressRouteCircuits/.test/common/deploy.test.bicep:83]: Is it intended to remove this line?

File: modules/Microsoft.Network/firewallPolicies/.test/common/deploy.test.bicep [hunk 14]
```diff
@@ -76,7 +76,14 @@ module testDeployment '../../deploy.bicep' = {
         ]
       }
     ]
+<<<<<<< HEAD
```
Comment (eriqua) [inline modules/Microsoft.Network/firewallPolicies/.test/common/deploy.test.bicep:79]: Please remove line
```suggestion
```

File: modules/Microsoft.Network/firewallPolicies/.test/common/deploy.test.bicep [hunk 15]
```diff
@@ -76,7 +76,14 @@ module testDeployment '../../deploy.bicep' = {
...
+    tags: {
+      Environment: 'Non-Prod'
+      Role: 'DeploymentValidation'
+    }
+=======
```
Comment (eriqua) [inline modules/Microsoft.Network/firewallPolicies/.test/common/deploy.test.bicep:84]: Please remove line
```suggestion
```

File: modules/Microsoft.Network/firewallPolicies/.test/common/deploy.test.bicep [hunk 16]
```diff
@@ -76,7 +76,14 @@ module testDeployment '../../deploy.bicep' = {
...
     allowSqlRedirect: true
     autoLearnPrivateRanges: 'Enabled'
+>>>>>>> 256b4b10ef1f210ecc80876c2d90dcdc5125d3bf
```
Comment (eriqua) [inline modules/Microsoft.Network/firewallPolicies/.test/common/deploy.test.bicep:87]: Please remove line
```suggestion
```

File: modules/Microsoft.Network/virtualNetworkGateways/.test/vpn/deploy.test.bicep [hunk 17]
```diff
@@ -87,6 +87,12 @@ module testDeployment '../../deploy.bicep' = {
       }
     ]
     vpnType: 'RouteBased'
+<<<<<<< HEAD
```
Comment (eriqua) [inline modules/Microsoft.Network/virtualNetworkGateways/.test/vpn/deploy.test.bicep:90]: Please remove line
```suggestion
```

File: modules/Microsoft.Network/virtualNetworkGateways/.test/vpn/deploy.test.bicep [hunk 18]
```diff
@@ -126,5 +132,6 @@ module testDeployment '../../deploy.bicep' = {
       }
     ]
     enableBgpRouteTranslationForNat: true
+>>>>>>> 256b4b10ef1f210ecc80876c2d90dcdc5125d3bf
```
Comment (eriqua) [inline modules/Microsoft.Network/virtualNetworkGateways/.test/vpn/deploy.test.bicep:135]: Please remove line
```suggestion
```

File: modules/Microsoft.Web/staticSites/.test/common/deploy.test.bicep [hunk 19]
```diff
@@ -72,6 +72,7 @@ module testDeployment '../../deploy.bicep' = {
         principalType: 'ServicePrincipal'
       }
     ]
+
```
Comment (eriqua) [inline modules/Microsoft.Web/staticSites/.test/common/deploy.test.bicep:75]: Please remove line
```suggestion
```

File: .ps-rule/dep-suppress.Rule.yaml [hunk 20]
```diff
@@ -11,3 +11,4 @@ spec:
       - 'dep'
       - 'ms.'
       - 'privatelink.'
+      - 'pe-'
```
Comment (eriqua) [inline .ps-rule/dep-suppress.Rule.yaml:14]: I'd suggest to remove this line. If dependencies are starting with a different suffix than "dep" we should rename them (in a different PR). Failing rules could be a useful way to track them.

File: .ps-rule/na-suppress.Rule.yaml [hunk 21]
```diff
@@ -0,0 +1,31 @@
+---
+# Synopsis: Suppress Rules for Not Available resources
```
Comment (eriqua) [inline .ps-rule/na-suppress.Rule.yaml:2]: Please list the new suppression in the docs, specifically this file https://github.com/Azure/ResourceModules/blob/main/docs/wiki/The%20CI%20environment%20-%20Pipeline%20design.md#psrule-pre-flight-validation-pipeline
In the PSRule section as a new bullet point in the suppression groups list

Comment (fabmas) [reply .ps-rule/na-suppress.Rule.yaml:2]: done.

File: modules/Microsoft.DesktopVirtualization/workspaces/.test/common/deploy.test.bicep (hunk 11 above)
Comment (elanzel) [reply modules/Microsoft.DesktopVirtualization/workspaces/.test/common/deploy.test.bicep:84]: workspaceFriendlyName removed

File: modules/Microsoft.Network/expressRouteCircuits/.test/common/deploy.test.bicep (hunk 13 above)
Comment (elanzel) [reply modules/Microsoft.Network/expressRouteCircuits/.test/common/deploy.test.bicep:83]: allowClassicOperations: true restored

**Review by eriqua — APPROVED** [review]
🆗

**Comment by BernieWhite** [comment]
There might be an typo with `Microsoft.ApiManagementservice/eventGridFilters` should be `Microsoft.ApiManagement/service/eventGridFilters`.

See: https://github.com/Azure/ResourceModules/blob/e16d91f8448a3a75a9c14f81cad5352f087c0f5e/.ps-rule/na-suppress.Rule.yaml#L30

**Pull request merged**